### ISSUE TITLE
CQL "date" type: No results

### ISSUE THREAD

**Issue by wasifaleem** [Stratio/cassandra-lucene-index#158]
Hi.

I am testing out a simplistic use case of indexing date fields, but I ran into a problem.

Consider the following example schema.

```cql
CREATE KEYSPACE IF NOT EXISTS test_date WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'};

CREATE TABLE test_date.test (
  id text PRIMARY KEY,
  dob date,
  lucene text
);

CREATE CUSTOM INDEX test_date_index ON test_date.test (lucene)
USING 'com.stratio.cassandra.lucene.Index'
WITH OPTIONS = {
'refresh_seconds' : '1',
'schema' : '{
        fields : {
            id    : {type : "string"},
            dob     : {type : "date", pattern : "yyyy-MM-dd"}
        }
    }'
};
```

The corresponding data and query:

```cql
INSERT INTO test_date.test (id, dob) VALUES ( 'unique_id', '2015-10-10');

SELECT * FROM test_date.test
WHERE lucene = '{filter : {
                       type  : "match",
                       field : "dob",
                       value : "2015-10-10" }}';

 id | dob | lucene
----+-----+--------

(0 rows)
```
The log file contains an error event
```log
INFO  [MemtablePostFlush:82] 2016-06-21 17:21:48,605 Index.java:239 - Flushed Lucene index test_date.test.test_date_index
ERROR [SharedPool-Worker-2] 2016-06-21 17:22:12,120 Schema.java:140 - Error in Lucene index:
        while mapping : Columns{id=unique_id, dob=-2147466930}
        with mapper   : DateMapper{field=dob, indexed=true, sorted=false, validated=false, column=dob, pattern=yyyy-MM-dd}
        caused by     : Required positive Integer for dates but found '-2147466930'
(END)

```

If I do the same, but with CQL type "timestamp", I do get proper results.
I am on versions 2.2.4 and 2.2.6.

**Comment by adelapena** [comment]
Hi @wasifaleem 

You are totally right, this is a bug affecting all versions. I have [just fixed it](https://github.com/Stratio/cassandra-lucene-index/commit/c26852d39224c31d77f8ed66d48d9ad2b5b983c3) at [3.0.7 branch](https://github.com/Stratio/cassandra-lucene-index/tree/branch-3.0.7), we will apply the patch to 2.2.x very soon.

Thanks for reporting

**Comment by ealonsodb** [comment]
This bug [patch](https://github.com/Stratio/cassandra-lucene-index/commit/c26852d39224c31d77f8ed66d48d9ad2b5b983c3) is included in releases [2.2.6.2](https://github.com/Stratio/cassandra-lucene-index/tree/2.2.6.2),[3.0.7.2](https://github.com/Stratio/cassandra-lucene-index/tree/3.0.7.2) and [3.7.1](https://github.com/Stratio/cassandra-lucene-index/tree/3.7.1)
Thanks
